pyfar/classes/audio.py

```diff
@@ -1497,87 +1512,82 @@ def _matrix_multiplication(a, b, axes, audio_type):
         a = np.expand_dims(a, 0) if a.ndim == 2 else a
         b = np.expand_dims(b, 1) if b.ndim == 2 else b
         # note: axes is implicitly copied
-        axes = [tuple([ax-1 if ax < 0 else ax for ax in t]) for t in axes]
+        axes = [tuple([ax - 1 if ax < 0 else ax for ax in t]) for t in axes]
     return np.matmul(a, b, axes=axes)
 
 
-def _match_fft_norm(fft_norm_1, fft_norm_2, division=False):
+def _match_fft_norm(fft_norm1, fft_norm2, operation, norm_override=None):
     """
-    Helper function to determine the fft_norm resulting from an
-    arithmetic operation of two audio objects.
-
-    For addition, subtraction and multiplication:
-    Either: one signal has fft_norm ``'none'`` , the results gets the other
-    norm.
-    Or: both have the same fft_norm, the results gets the same norm.
-    Other combinations raise an error.
-
-    For division:
-    Either: the denominator (fft_norm_2) is ``'none'``, the result gets the
-    fft_norm of the numerator (fft_norm_1).
-    Or: both have the same fft_norm, the results gets the fft_norm ``'none'``.
-    Other combinations raise an error.
+    Determine the appropriate FFT normalization after an arithmetic operation
+    based on the provided norms of the two input signals.
 
     Parameters
     ----------
-    fft_norm_1 : str, ``'none'``, ``'unitary'``, ``'amplitude'``, ``'rms'``,
-    ``'power'`` or ``'psd'``
-        First fft_norm for matching.
-    fft_norm_2 : str, ``'none'``, ``'unitary'``, ``'amplitude'``, ``'rms'``,
-    ``'power'`` or ``'psd'``
-        Second fft_norm for matching.
-    division : bool
-        ``False`` if arithmetic operation is addition, subtraction or
-        multiplication;
-        ``True`` if arithmetic operation is division.
-
+    fft_norm1 : str
+        The FFT normalization of the first signal. Must be one of 'none',
+        'unitary', 'amplitude', 'rms', 'power', or 'psd'.
+    fft_norm2 : str
+        The FFT normalization of the second signal. Must adhere to the same
+        constraints as `fft_norm1`.
+    operation : str
+        The arithmetic operation to be performed. Supported operations are
+        'multiply', 'divide', 'add', or 'subtract'.
+    norm_override : str, optional
+        A specific normalization to override any default behavior. Must be one
+        of the valid norms if specified.
     Returns
     -------
-    fft_norm_result : str, ``'none'``, ``'unitary'``, ``'amplitude'``,
-    ``'rms'``, ``'power'`` or ``'psd'``
-        The fft_norm resulting from arithmetic operation.
-    """
+    str
+        The resulting FFT normalization after considering the operations and
+        input norms.
+    Raises
+    ------
+    UserWarning
+        Warns about the use of an invalid norm, unsupported operations,
+        mismatched norms without override, or specific concerns
+        about operations involving 'none' normalization.
```

Comment:
We actually want to be strict here and raise errors, not warning in case something is not as defined in our rules for arithmetics.